great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -138,6 +134,158 @@ def test_connection(self, table_asset: TableAsset) -> None:
+
+    def splitter_method_kwargs(self) -> Dict[str, Any]:
+        return {"column_name": self.column_name, "date_parts": self.param_names}
+
+    def request_options_to_batch_spec_kwarg_identifiers(
+        self, options: BatchRequestOptions
+    ) -> Dict[str, Any]:
+        return _request_options_to_datetime_batch_spec_kwargs(self, options)
+
+    @pydantic.validator("datetime_parts", each_item=True)
+    def _check_param_name_allowed(cls, v: str):
+        from great_expectations.execution_engine.split_and_sample.data_splitter import (
+            DatePart,
+        )
+
+        allowed_date_parts = [part.value for part in DatePart]
+        assert (
+            v in allowed_date_parts
+        ), f"Only the following param_names are allowed: {allowed_date_parts}"
+        return v
+
+
+DatetimeColumnSplitter = Union[
+    ColumnSplitterYear,
+    ColumnSplitterYearAndMonth,
+    ColumnSplitterYearAndMonthAndDay,
+    ColumnSplitterDatetimePart,
+]
+
+
+def _request_options_to_datetime_batch_spec_kwargs(
+    splitter: DatetimeColumnSplitter, options: BatchRequestOptions
+) -> Dict[str, Any]:
+    identifiers: Dict = {}
+    for part in splitter.param_names:
+        if part not in options:
+            raise ValueError(
+                f"'{part}' must be specified in the batch request options to create a batch identifier"
+            )
+        identifiers[part] = options[part]
+    return {splitter.column_name: identifiers}
+
+
+class ColumnSplitterColumnValue(_ColumnSplitter):
+    method_name: Literal["split_on_column_value"] = "split_on_column_value"
+
+    @property
+    def param_names(self) -> List[str]:
+        return [self.column_name]
+
+    def splitter_method_kwargs(self) -> Dict[str, Any]:
+        return {"column_name": self.column_name}
+
+    def request_options_to_batch_spec_kwarg_identifiers(
+        self, options: BatchRequestOptions
+    ) -> Dict[str, Any]:
+        if self.column_name not in options:
+            raise ValueError(
+                f"'{self.column_name}' must be specified in the batch request options to create a batch identifier"
+            )
+        return {self.column_name: options[self.column_name]}
+
+
+class ColumnSplitterDividedInteger(_ColumnSplitter):
+    divisor: int
+    method_name: Literal["split_on_divided_integer"] = "split_on_divided_integer"
+
+    @property
+    def param_names(self) -> List[str]:
+        return ["quotient"]
+
+    def splitter_method_kwargs(self) -> Dict[str, Any]:
+        return {"column_name": self.column_name, "divisor": self.divisor}
+
+    def request_options_to_batch_spec_kwarg_identifiers(
+        self, options: BatchRequestOptions
+    ) -> Dict[str, Any]:
+        if "quotient" not in options:
+            raise ValueError(
+                "'quotient' must be specified in the batch request options to create a batch identifier"
+            )
+        return {self.column_name: options["quotient"]}
+
+
+ColumnSplitter = Union[
```

Comment:
@billdirks There seems to be some overlap between the members of `ColumnSplitter` and `DatetimeColumnSplitter`; would it be useful to somehow combine them in a purely composable way?  Thanks.

Reply:
These are distinct types. How do you propose we compose them? Note a union of union types is not equivalent to unioning the subtypes together.